test/issues/99_canonical_with_junction_point.cpp

```diff
@@ -0,0 +1,70 @@
+//  Boost operations_test.cpp  ---------------------------------------------------------//
+
+//  Copyright Alexander Grund 2020
+
+//  Distributed under the Boost Software License, Version 1.0.
+//  See http://www.boost.org/LICENSE_1_0.txt
+
+//  Library home page: http://www.boost.org/libs/filesystem
+
+#include <boost/filesystem.hpp>
+#include <boost/core/lightweight_test.hpp>
+#include <iostream>
+#include <vector>
+
+namespace fs = boost::filesystem;
+
+struct TmpDir
+{
+  fs::path path;
+  TmpDir(const fs::path& base): path(fs::absolute(base) / fs::unique_path())
+  {
+    fs::create_directories(path);
+  }
+  ~TmpDir()
+  {
+    boost::system::error_code ec;
+    fs::remove_all(path, ec);
+  }
+};
+
+// Test fs::canonical for various path in a Windows directory junction point
+// This failed before due to broken handling of absolute paths and ignored ReparseTag
+int main()
+{
+  if (std::system("mklink /?") != 0)
```

Comment:
This check doesn't work on Windows 8.1. `mklink` exists, it prints help, and then the test decides it doesn't exist and exits.

Reply:
I don't know why `std::system` returns 1, but it does. It also returns 1 if the file doesn't exist. I think `std::system` result is not indicative and not trustworthy, we need to find another way. Possibly, from the Jamfile.

Reply:
Maybe try running it and check if the link got created. Only skip if it wasn't. 
Also the test should include `mklink /D ...` The failures were different with that type of link. And there could be differences whether it's an absolute or relative link to absolute or relative path.

Reply:
I've already implemented a different check in the Jamfile.

Reply:
Another problem with this test is that it doesn't remove the temporary directory it creates. `remove_all` fails at removing "<temp_dir>/real/sub" with error 32 (directory used by another process).